resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/PodBuilderSuite.scala

```diff
@@ -249,3 +283,57 @@ class TestStepTwo extends KubernetesFeatureConfigStep {
     SparkPod(podWithLocalDirVolumes, containerWithLocalDirVolumeMounts)
   }
 }
+
+/**
+ * A test user feature step.
+ */
+class TestStepWithConf extends KubernetesFeatureConfigStep {
+  import io.fabric8.kubernetes.api.model._
+
+  private var kubernetesConf: KubernetesConf = _
+
+  override def init(conf: KubernetesConf): Unit = {
+    kubernetesConf = conf
+  }
```

Comment:
If user want to use `KubernetesConf` rather than driver or executor conf, they need to reimplements `KubernetesFeatureConfigStep.init`